source/samples/pubsub/PubSubFeature.cpp

```diff
@@ -30,6 +34,54 @@ string PubSubFeature::getName()
     return "Pub Sub Sample";
 }
 
+bool PubSubFeature::createPubSub(const PlainConfig &config, std::string absFilePath)
+{
+    std::string pubFileDir = FileUtils::ExtractParentDirectory(absFilePath);
+    if (!FileUtils::DirectoryExists(pubFileDir))
+    {
+        // Create an empty directory with the expected permissions.
+        FileUtils::CreateDirectoryWithPermissions(pubFileDir.c_str(), S_IRWXU | S_IRGRP | S_IROTH | S_IXOTH);
+    }
+    else
+    {
+        // Verify the directory permissions.
+        auto rcvDirPermissions = FileUtils::GetFilePermissions(pubFileDir);
+        if (Permissions::PUBSUB_DIR != rcvDirPermissions)
+        {
+            LOGM_ERROR(
+                TAG,
+                "Incorrect directory permissions for pubsub file: %s expected: %d received: %d",
+                Sanitize(pubFileDir).c_str(),
+                Permissions::PUBSUB_DIR,
+                rcvDirPermissions);
+            return false;
+        }
+    }
+
+    if (!FileUtils::FileExists(absFilePath))
+    {
+        // Create an empty file with the expected permissions.
+        FileUtils::CreateEmptyFileWithPermissions(absFilePath, S_IRUSR | S_IWUSR);
```

Comment:
This function returns `true/false` on `success/failure`. You have to test for return and propagate back to caller.

```
if (!FileUtils::CreateEmptyFileWithPermissions(absFilePath, S_IRUSR | S_IWUSR))
{
    return false;
}
```

Reply:
done